common/djangoapps/dark_lang/api.py

```diff
@@ -0,0 +1,199 @@
+
+        # Set the session key to the requested preview lang
+        request.session[LANGUAGE_SESSION_KEY] = preview_lang
+
+        # Make sure that we set the requested preview lang as the dark lang preference for the
+        # user, so that the lang_pref middleware doesn't clobber away the dark lang preview.
+        if auth_user:
+            set_user_preference(request.user, DARK_LANGUAGE_KEY, preview_lang)
+        return 'Language set to %s' % preview_lang
+
+    def clear_preview_language(self, request):
+        """
+        Clears the dark language preview
+        :param request: The incoming Django Request
+        :return: The Response message
+        """
+        auth_user = request.user.is_authenticated()
+
+        # delete the session language key (if one is set)
+        if LANGUAGE_SESSION_KEY in request.session:
+            del request.session[LANGUAGE_SESSION_KEY]
+
+        user_pref = None
+        if auth_user:
+            # Reset user's dark lang preference to null
+            delete_user_preference(request.user, DARK_LANGUAGE_KEY)
+            # Get & set user's preferred language
+            user_pref = get_user_preference(request.user, LANGUAGE_KEY)
+            if user_pref:
+                request.session[LANGUAGE_SESSION_KEY] = user_pref
+        return user_pref
+
+    def fuzzy_match(self, lang_code):
+        """Returns a fuzzy match for lang_code"""
+        if lang_code in self.released_langs:
+            return lang_code
+
+        lang_prefix = lang_code.partition('-')[0]
+        for released_lang in self.released_langs:
+            released_prefix = released_lang.partition('-')[0]
+            if lang_prefix == released_prefix:
+                return released_lang
+        return None
+
+    def dark_parse_accept_lang_header(self, accept):
+        """
+        The use of 'zh-cn' for 'Simplified Chinese' and 'zh-tw' for 'Traditional Chinese'
+        are now deprecated, as discussed here: https://code.djangoproject.com/ticket/18419.
+        The new language codes 'zh-hans' and 'zh-hant' are now used since django 1.7.
+        Although majority of browsers still use the old language codes, some new browsers
+        such as IE11 in Windows 8.1 start to use the new ones, which makes the current
+        chinese translations of edX don't work properly under these browsers.
+        This function can keep compatibility between the old and new language codes. If one
+        day edX uses django 1.7 or higher, this function can be modified to support the old
+        language codes until there are no browsers use them.
+        """
+        browser_langs = parse_accept_lang_header(accept)
+        django_langs = []
+        for lang, priority in browser_langs:
+            lang = CHINESE_LANGUAGE_CODE_MAP.get(lang.lower(), lang)
+            django_langs.append((lang, priority))
+
+        return django_langs
+
+    def _format_accept_value(self, lang, priority=1.0):
```

Comment:
I would just inline the code of this method where it is used, given that it is only used in a single place and it is so short.